bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -234,10 +234,10 @@ public Rule add(Rule rule) {
         String rUID = rule.getUID();
         if (rUID == null) {
             rUID = ruleEngine.getUniqueId();
-            super.add(initRuleId(rUID, rule));
-        } else {
-            super.add(rule);
+            rule = initRuleId(rUID, rule);
         }
+        ruleEngine.normalizeRuleConfigurations(rule);
```

Comment:
With Things, we indeed try to normalize them as early as possible, i.e. once we get hold of the configuration for the first time (e.g. in the `@PUT` annotated methods in REST resources or right after reading the DSLs files from disk. The goal really is to have only normalized configurations in the ThingRegistry.